client/dashboard/app/queries/domains.ts

```diff
@@ -16,3 +18,11 @@ export const domainSuggestionsQuery = (
 		queryKey: [ 'domain-suggestions', search, domainSuggestionQuery ],
 		queryFn: () => fetchDomainSuggestions( search, domainSuggestionQuery ),
 	} );
+
+export const updateDNSSECMutation = ( domain: string, siteId: number ) =>
```

Comment:
[nit] we tend to name mutations in this pattern (note the `domain` prefix); see similar pattern in site-related queries / mutations


```suggestion
export const domainDnssecMutation = ( domain: string, siteId: number ) =>
```